### PR TITLE
perms-syncer: Add test for nil provider but with linked user ids 

### PR DESCRIPTION
**Pull request by indradhanush**
We fixed a panic with nil provider object in f018333. This commit
adds a test to ensure it doesn't happen again.

### CODE REVIEW

File: enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go [hunk 1]
```diff
@@ -567,6 +567,51 @@ func TestPermsSyncer_syncRepoPerms(t *testing.T) {
...
+			}
+
+			wantUserIDs := []uint32{1}
+			if diff := cmp.Diff(wantUserIDs, p.UserIDs.ToArray()); diff != "" {
+				return errors.Errorf("UserIDs mismatch (-want +got):\n%s", diff)
+			}
+			return nil
+		}
+		edb.Mocks.Perms.SetRepoPendingPermissions = func(ctx context.Context, accounts *extsvc.Accounts, p *authz.RepoPermissions) error {
+			return nil
```
Comment (unknwon) [inline enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go:589]: nit: this method shouldn't be called due to no provider, we could add an assert here.

**Review by unknwon — APPROVED** [review]
LGTM! Assume tested with undo https://github.com/sourcegraph/sourcegraph/commit/f018333c31dbc43a602dcba44034b9a140729888 and observe a panic.

**Comment by indradhanush** [comment]
Hadn't tested without the fix, but had manually stepped through the code with `dlv` to ensure this worked. Good idea nonetheless. Just ran it locally. Here it is for posterity's sake:

```
 $ go test -run TestPermsSyncer_syncRepoPerms
--- FAIL: TestPermsSyncer_syncRepoPerms (0.00s)
    --- FAIL: TestPermsSyncer_syncRepoPerms/repo_sync_with_external_service_userid_but_no_providers (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xea5373]

goroutine 30 [running]:
testing.tRunner.func1.2(0xfaa720, 0x19134c0)
	/usr/lib/go/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000542300)
	/usr/lib/go/src/testing/testing.go:1146 +0x4b6
panic(0xfaa720, 0x19134c0)
	/usr/lib/go/src/runtime/panic.go:965 +0x1b9
github.com/sourcegraph/sourcegraph/enterprise/cmd/repo-updater/internal/authz.(*PermsSyncer).syncRepoPerms(0xc00038e600, 0x12ab9d0, 0xc00038e6f0, 0x1, 0xc00049fe10, 0x611a698b)
	/home/dhanush/sourcegraph/sourcegraph/enterprise/cmd/repo-updater/internal/authz/perms_syncer.go:579 +0x993
github.com/sourcegraph/sourcegraph/enterprise/cmd/repo-updater/internal/authz.TestPermsSyncer_syncRepoPerms.func4(0xc000542300)
	/home/dhanush/sourcegraph/sourcegraph/enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go:609 +0x11d
testing.tRunner(0xc000542300, 0xc00054e1b0)
	/usr/lib/go/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/usr/lib/go/src/testing/testing.go:1238 +0x2b3
exit status 2
FAIL	github.com/sourcegraph/sourcegraph/enterprise/cmd/repo-updater/internal/authz	0.019s
```

File: enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go (hunk 1 above)
Comment (indradhanush) [reply enterprise/cmd/repo-updater/internal/authz/perms_syncer_test.go]: @unknwon Updated the mock to return an error. How does it look now?

**Review by unknwon — APPROVED** [review]
Thanks!

**Pull request merged**